src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs

```diff
@@ -0,0 +1,109 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Xml;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+
+namespace Microsoft.Playwright.TestAdapter
+{
+    [ExtensionUri("settings://playwright")]
+    [SettingsName("Playwright")]
+    public class PlaywrightSettingsProvider : ISettingsProvider
+    {
+        private static PlaywrightSettingsXml? _settings = null!;
+
+        public static string BrowserName
+        {
+            get
+            {
+                var browserFromEnv = Environment.GetEnvironmentVariable("BROWSER")?.ToLower();
+                if (browserFromEnv != null)
+                {
+                    ValidateBrowserName(browserFromEnv);
+                    return browserFromEnv;
+                }
+                if (_settings != null && !string.IsNullOrEmpty(_settings.Browser))
+                {
+                    var browser = _settings.Browser!.ToLower();
+                    ValidateBrowserName(browser);
+                    return browser;
+                }
+                return BrowserType.Chromium;
+            }
+            private set { }
+        }
+
+        public static float? ExpectTimeout
+        {
+            get
+            {
+                if (_settings == null)
```

Comment:
in ALL the getters `_settings` can be null if there is no `RUNSETTINGS` file given. We fallback then to defaults.